1.setup_TD-VAE/scripts/0.setup_TD-VAE.py

```diff
@@ -0,0 +1,214 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[8]:
+
+
+import pathlib
+import pickle
+import sys
+
+import matplotlib.pyplot as plt
+import numpy as np
+import optuna
+import pandas
+import torch
+import tqdm
+from matplotlib import gridspec
+from torch import nn, optim
+from torch.nn import functional as F
+from torch.utils.data import DataLoader
+
+utils_path = pathlib.Path("../../utils/").resolve(strict=True)
+sys.path.append(str(utils_path))
+
+from model import TD_VAE, DBlock, Decoder, PreProcess
+from prep_data import MNIST_Dataset
+from rollout import rollout_func
+
+# In[9]:
+
+
+# set path to the MNIST images
+mnist_pickle_path = pathlib.Path("../../data/mnist/MNIST.pkl").resolve(strict=True)
+# create the log directory if it does not exist
+log_path = pathlib.Path("../log/").resolve()
+log_path.mkdir(exist_ok=True)
+log_file_path = pathlib.Path("../log/loginfo.txt").resolve()
```

Comment:
I recommend using Python's logger, as it can capture silenced debugging messages from other packages.

Reply:
added!